tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/Services/FailureAnalysis/GitCheckoutFailureClassifier.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Azure.Sdk.Tools.PipelineWitness.Services.FailureAnalysis
+{
+    public class GitCheckoutFailureClassifier : IFailureClassifier
+    {
+        public async Task ClassifyAsync(FailureAnalyzerContext context)
+        {
+            var tasks = from r in context.Timeline.Records
+                       where r.RecordType == "Task"
+                       where r.Issues.Any(i => i.Message.Contains("Git fetch failed with exit code: 128"))
```

Comment:
Does this also find cases were DevOps hits this while trying to retrieve the yml file to see if the pipeline should be triggered at all? I'm not sure if they have those errors surface in the same way or not.